operator/pkg/gateway-api/controller.go

```diff
@@ -188,6 +189,59 @@ func getGatewaysForSecret(ctx context.Context, c client.Client, obj client.Objec
 	return gateways
 }
 
+func getGatewaysForNamespace(ctx context.Context, c client.Client, ns client.Object) []types.NamespacedName {
```

Comment:
adding a unittest? these mapping functions are quite easy to test - even though one might need the fakek8sclient.

Reply:
ah actually, I already have unit test for getGatewaysForSecrets :)